pkg/manual_tests/user_default_database_and_role/test.md

```diff
@@ -0,0 +1,137 @@
+# User default database and role
+
+This test shows how Snowflake behaves whenever you log into a user that has default namespace (database) and default role set up.
+The cases are covering the situations where:
+- Either database or role is not present.
+- Database and role are present, but not granted to the user.
+- Database and role are granted to the user, but the casing is not matched.
+- Database and role are granted to the user and the casing is matching.
+- Only role is granted to the user and the casing is matching.
+
+## Setup
+
+As a testing environment, I chose VSCode with the Snowflake extension for being able to quickly go between ACCOUNTADMIN account
+and the tested one. For testing account, I added this configuration for the tested user:
+
+```toml
+[connections.Test]          
+accountname = '***'
+username = '***'
+password= '***'
+host = '***'
+```
+
+The only thing that won't change between the tests is role and database, so we can create it beforehand:
+
+```sql
+CREATE ROLE TEST_ROLE;
+CREATE DATABASE TEST_DATABASE;
+```
+
+> Note: We have to use TYPE = LEGACY_SERVICE to be able to use 
+
+### 1. Either database or role is not present.
+
+1. Create user with the non-existing default database and role.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'NON_EXISTING_TEST_DATABASE'
+    DEFAULT_ROLE = 'NON_EXISTING_TEST_ROLE';
+```
+
+2. Log into the user (logs in successfully, but no database and role is selected in the context).
+
+### 2. Database and role are present, but not granted to the user.
+
+1. Replace the user with existing default database and role.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'TEST_DATABASE'
+    DEFAULT_ROLE = 'TEST_ROLE';
+```
+
+2. Log into the user (logs in successfully, but no database and role is selected in the context).
+   
+### 3. Database and role are granted to the user, but the casing is not matched.
+
+1. Replace the user with existing default database and role, but with lowercase names.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'test_database'
```

Comment:
Is our SDK using single or double quotes here? (I am almost sure that the double ones - maybe this is not a difference in behavior but wanted to be sure)

Reply:
I tested it in a few ways and it didn't make any difference. I'll leave it as is and add note at the top that the quoting doesn't matter in this case.